app/models/goal.py

```diff
@@ -2,4 +2,28 @@
 
 
 class Goal(db.Model):
-    goal_id = db.Column(db.Integer, primary_key=True)
+    goal_id = db.Column(db.Integer, primary_key=True,  autoincrement = True)
+    title = db.Column(db.String)
+    tasks = db.relationship("Task", back_populates = "goal")
+
+    @classmethod
+    def from_json(cls, req_body):
+        return cls(
+                title = req_body["title"]
+        )
+
+    def to_dict(self):
+        return(
+            {
+                "goal": {
+                    "id": self.goal_id,
+                    "title": self.title
+                }
+            }
```

Comment:
I would consider refactoring this, so that it returns only the model instance we are casting to a dictionary.

I think that will make this method more flexible, robust, and reusable. Anytime you _do_ need to surround it with a "goal" key, you can do that elsewhere (maybe in the route when you are creating the response body).

Also, how could we change this, so that if the goal has tasks, we can also add them into a list and return it?